src/Workspaces/Core/Portable/Workspace/Solution/Checksum.cs

```diff
@@ -36,6 +35,14 @@ internal sealed partial class Checksum : IObjectWritable, IEquatable<Checksum>
         public Checksum(HashData hash)
             => _checksum = hash;
 
+        internal ImmutableArray<byte> ToImmutableArray()
+        {
+            Span<byte> data = stackalloc byte[HashSize];
+            Contract.ThrowIfFalse(MemoryMarshal.TryWrite(data, ref Unsafe.AsRef(in _checksum)));
+            var arrayData = data.ToArray();
```

Comment:
since we're making an array anyways to feed into DangerousCreateFromUnderlyingArray, do we need the SPan/Stackalloc?  Could we not just do `new byte[ HashSize]` and then just have hte Hashdata write directly into that?  Seems liek it would avoid a copy from the locak stack to the array. (likely very minor, but it seems a but simpler).

Reply:
➡️ Fixed